File: src/C3.Infrastructure/Preferences/UserPreferencesSnapshot.vb
Imports C3.Infrastructure.Updates

Namespace Preferences

Public NotInheritable Class UserPreferencesSnapshot

    Public Const MaximumDefaultDirectoryCharacters As Integer = 32768
    Public Const CurrentLegacyImportVersion As Integer = 1
    Public Const ImportOutcomePending As String = "pending"
    Public Const ImportOutcomeImported As String = "imported"
    Public Const ImportOutcomeNotFound As String = "not-found"
    Public Const ImportOutcomeInvalid As String = "invalid"

    Public Property ShowMessages As Boolean

    Public Property DefaultDirectory As String

    Public Property UpdatePolicy As UpdateCheckPolicy

    Public Property LastUpdateCheck As DateTime

    Public Property Legacy1xImportVersion As Integer

    Public Property Legacy1xImportOutcome As String

    Public Shared Function CreateDefaults(myDocumentsPath As String) As UserPreferencesSnapshot
        Return New UserPreferencesSnapshot() With {
            .ShowMessages = True,
            .DefaultDirectory = If(myDocumentsPath, String.Empty),
            .UpdatePolicy = UpdateCheckPolicy.Never,
            .LastUpdateCheck = DateTime.MinValue,
            .Legacy1xImportVersion = 0,
            .Legacy1xImportOutcome = ImportOutcomePending
        }
    End Function

    Public Function Clone() As UserPreferencesSnapshot
        Return New UserPreferencesSnapshot() With {
            .ShowMessages = ShowMessages,
            .DefaultDirectory = DefaultDirectory,
            .UpdatePolicy = UpdatePolicy,
            .LastUpdateCheck = LastUpdateCheck,
            .Legacy1xImportVersion = Legacy1xImportVersion,
            .Legacy1xImportOutcome = Legacy1xImportOutcome
        }
    End Function

End Class

End Namespace

File: src/C3.Infrastructure/Preferences/UserPreferencesSnapshot.cs
using C3.Infrastructure.Updates;
using System;
using System.Runtime.CompilerServices;

namespace C3.Infrastructure.Preferences
{
	public sealed class UserPreferencesSnapshot
	{
		public const int MaximumDefaultDirectoryCharacters = 32768;

		public const int CurrentLegacyImportVersion = 1;

		public const string ImportOutcomePending = "pending";

		public const string ImportOutcomeImported = "imported";

		public const string ImportOutcomeNotFound = "not-found";

		public const string ImportOutcomeInvalid = "invalid";
		public bool ShowMessages
		{
			get;
			set;
		}

		public string DefaultDirectory
		{
			get;
			set;
		}

		public UpdateCheckPolicy UpdatePolicy
		{
			get;
			set;
		}

		public DateTime LastUpdateCheck
		{
			get;
			set;
		}

		public int Legacy1xImportVersion
		{
			get;
			set;
		}

		public string Legacy1xImportOutcome
		{
			get;
			set;
		}

		public static UserPreferencesSnapshot CreateDefaults(string myDocumentsPath)
		{
			return new UserPreferencesSnapshot
			{
				ShowMessages = true,
				DefaultDirectory = (myDocumentsPath ?? string.Empty),
				UpdatePolicy = UpdateCheckPolicy.Never,
				LastUpdateCheck = DateTime.MinValue,
				Legacy1xImportVersion = 0,
				Legacy1xImportOutcome = "pending"
			};
		}

		public UserPreferencesSnapshot Clone()
		{
			return new UserPreferencesSnapshot
			{
				ShowMessages = ShowMessages,
				DefaultDirectory = DefaultDirectory,
				UpdatePolicy = UpdatePolicy,
				LastUpdateCheck = LastUpdateCheck,
				Legacy1xImportVersion = Legacy1xImportVersion,
				Legacy1xImportOutcome = Legacy1xImportOutcome
			};
		}
	}
}
